src/test/java/com/google/sps/servlets/LiveCourseDataTest.java

```diff
@@ -0,0 +1,108 @@
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+public final class LiveCourseDataTest {
+  private static final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  private AddSchoolData newSchoolObject;
+  private LiveCourseData LiveData;
+
+  @BeforeEach
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+    newSchoolObject = new AddSchoolData();
+    LiveData = new LiveCourseData();
+  }
+
+  @AfterEach
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Mock HttpServletRequest request;
+  @Mock HttpServletRequest requestB;
+
+  @Test
+  public void FindingExisitngTermEntity() {
+    DatastoreService db = DatastoreServiceFactory.getDatastoreService();
+    request = createRequest(request, "MIT", "6.006", "Spring 2020", "12", "Jason Ku");
+    requestB = createRequest(requestB, "MIT", "6.008", "Spring 2018", "6", "Srini");
+    String expectedTermName = "Spring 2020";
+
+    newSchoolObject.addSchoolData(db, request);
+    newSchoolObject.addSchoolData(db, requestB);
+    Key expectedParent = findQueryMatch(db, "Course", "course-name", "6.006").get(0).getKey();
+    Entity found = LiveData.findData(db, request);
+
+    assertEquals(expectedParent, found.getParent());
+    assertEquals(expectedTermName, found.getProperty("term"));
+  }
+
+  private HttpServletRequest createRequest(
+      HttpServletRequest request,
+      String schoolName,
+      String courseName,
+      String termName,
+      String units,
+      String profName) {
+    request = Mockito.mock(HttpServletRequest.class);
+    when(request.getParameter("school-name")).thenReturn(schoolName);
+    when(request.getParameter("course-name")).thenReturn(courseName);
+    when(request.getParameter("term")).thenReturn(termName);
+    when(request.getParameter("units")).thenReturn(units);
+    when(request.getParameter("professor-name")).thenReturn(profName);
+    return request;
+  }
+
+  private List<Entity> findQueryMatch(
+      DatastoreService db, String entityType, String entityProperty, String propertyValue) {
+    Filter filter = new FilterPredicate(entityProperty, FilterOperator.EQUAL, propertyValue);
+    Query q = new Query(entityType).setFilter(filter);
+    List<Entity> result = db.prepare(q).asList(FetchOptions.Builder.withDefaults());
+    return result;
+  }
+
+  private List<Entity> findQueryMatch(DatastoreService db, String entityType) {
+    Query q = new Query(entityType);
+    List<Entity> result = db.prepare(q).asList(FetchOptions.Builder.withDefaults());
+    return result;
+  }
```

Comment:
Is the second function necessary? Or is it  a duplicate of the first

Reply:
Not necessary I was using it to debug something.